packages/utils/node/fetch-global-access/lib/fetch-global-access-for-user.ts

```diff
@@ -0,0 +1,15 @@
+import type { Accountability, GlobalAccess } from '@directus/types';
+import type { Knex } from 'knex';
+import { fetchGlobalAccessForQuery } from '../utils/fetch-global-access-for-query.js';
+
+interface FetchGlobalAccessForUserContext {
+	knex: Knex;
+}
+
+export async function fetchGlobalAccessForUser(
+	accountability: Pick<Accountability, 'user' | 'ip'>,
+	context: FetchGlobalAccessForUserContext,
+): Promise<GlobalAccess> {
```

Comment:
Since this function and the one it calls doesnt use the complete accountability object do we want to expose the function to end users with its current signature? 🤔 

Instead of forcing the user to create a fake accountability object to use this utility we could have the signature be more strict:

```suggestion
export async function fetchGlobalAccessForUser(
	user: string, // dont think null has a use-case here?
	context: FetchGlobalAccessForUserContext & { ip?: string }, // `ip` optional as its not required in all cases
): Promise<GlobalAccess> {
```